backend/modules/parsers/multimodalparser.py

```diff
@@ -92,52 +108,41 @@ def __init__(
 
     async def call_vlm_agent(
         self,
+        llm: BaseChatModel,
         base64_image: str,
         page_number: int,
         prompt: str = "Describe the information present in the image in a structured format.",
     ):
-        logger.info(f"Processing Image... {page_number}")
-        headers = {
-            "Content-Type": "application/json",
-            "Authorization": f"Bearer {settings.TFY_API_KEY}",
-            # Set the tfy_log_request to "true" in X-TFY-METADATA header to log prompt and response for the request
-            "X-TFY-METADATA": json.dumps(
-                {"tfy_log_request": "true", "Custom-Metadata": "Custom-Value"}
-            ),
-        }
-        # Parallel calls to VLM
-        async with aiohttp.ClientSession() as session:
-            payload = {
-                "model": self.vision_model,
-                "messages": [
+        content = [
+            HumanMessage(
+                [
+                    {
+                        "type": "text",
+                        "text": prompt,
+                    },
                     {
-                        "role": "user",
-                        "content": [
-                            {"type": "text", "text": prompt},
-                            {
-                                "type": "image_url",
-                                "image_url": {
-                                    "url": f"data:image/jpeg;base64,{base64_image}"
-                                },
-                            },
-                        ],
-                    }
-                ],
-                "max_tokens": 2048,
+                        "type": "image_url",
+                        "image_url": {
+                            "url": f"data:image/jpeg;base64,{base64_image}",
+                            "detail": "high",
+                        },
+                    },
+                ]
+            )
+        ]
+        logger.info(f"Processing Image... {page_number}")
+
+        try:
+            response = await llm.ainvoke(content)
+            return {
+                "response": [
+                    page_number,
+                    response.content,
+                ]
             }
-            async with session.post(
-                self.client, json=payload, headers=headers
-            ) as response:
-                result = await response.json(content_type=response.content_type)
-                if "choices" in result:
-                    return {
-                        "response": [
-                            page_number,
-                            result["choices"][0]["message"]["content"],
-                        ]
-                    }
-                if "error" in result:
-                    return {"error": result["error"]}
+        except Exception as e:
+            logger.exception(f"Error in page: {page_number} - {e}")
+            return {"error": e}
```

Comment:
`e` may or may not be directly serializable

Reply:
Resolved.

Reply:
How was this resolved? I see the `return {"error": e}` is still as it is

Reply:
The idea behind this was to mostly to skip pages that had errors but, yes, we can halt the parsing altogether.

Reply:
Maybe this comment is not needed, I had a misunderstanding that this was returning this from get_chunks
I see the handling for this already exists at line 217